src/server/conn_context.h

```diff
@@ -196,6 +197,8 @@ class ConnectionContext : public facade::ConnectionContext {
   // Reference to a FlowInfo for this connection if from a master to a replica.
   FlowInfo* replication_flow;
 
+  acl::UserRegistry* user_registry;
```

Comment:
Why do you pass it around / store it everywhere?

Reply:
Main service owns it and will use it for security checks with fast access. The only commands that will access it directly are ACL commands

You can make the registry store itself in a global variable. IIRC our cluster config does so. Alternatively you can add pointers to... ServerState for example and then fetch it with ServerState::tlocal()->acl_user_registry

Reply:
let me check, I like this. Besides any check for the registry will be done at the `DispatchCommand`

Reply:
not a big fan of global variables so I ended up doing it via `ServerState`